sanic_jwt/initialization.py

```diff
@@ -0,0 +1,174 @@
+
+        self.app = app
+        self.bp = bp
+        self.kwargs = kwargs
+        self.instance = instance
+
+        self.__load_configuration()
+        self.__load_response()
+        self.__check_initialization()
+        self.__add_class_views()
+        self.__add_endpoints()
+        self.__initialize_instance()
+
+    def __add_endpoints(self):
+        """
+        Initialize the Sanic JWT Blueprint and add to the instance initialized
+        """
+        config = self.config
+
+        endpoint_mappings = (
+            ('AuthenticateEndpoint', 'authenticate'),
+            ('RetrieveUserEndpoint', 'retrieve_user'),
+            ('VerifyEndpoint', 'verify'),
+            ('RefreshEndpoint', 'refresh'),
+        )
+
+        for endpoint in endpoint_mappings:
+            self.__add_single_endpoint(*endpoint)
+
+        self.bp.exception(exceptions.SanicJWTException)(
+            self.response.exception_response)
+
+        if not self.instance_is_blueprint:
+            self.instance.blueprint(
+                self.bp, url_prefix=config.url_prefix)
+
+    def __add_class_views(self):
+        """
+        Include any custom class views on the Sanic JWT Blueprint
+        """
+        config = self.config
+        if 'class_views' in self.kwargs:
+            class_views = self.kwargs.pop('class_views')
+
+            for route, view in class_views:
+                if issubclass(view, HTTPMethodView) and isinstance(route, str):
+                    self.bp.add_route(
+                        view.as_view(),
+                        route,
+                        strict_slashes=config.strict_slashes
+                    )
+                else:
+                    raise exceptions.InvalidClassViewsFormat()
+
+    def __check_initialization(self):
+        """
+        Confirm that required parameters were initialized and report back exceptions
+        """
+        config = self.config
+        if hasattr(config, 'refresh_token_enabled') and \
+            getattr(config, 'refresh_token_enabled') and (
+            not self.kwargs.get('store_refresh_token') or
+            not self.kwargs.get('retrieve_refresh_token')
+        ):
+            raise exceptions.RefreshTokenNotImplemented
+
+        # TODO:
+        # - Add additional checks
+
+    def __initialize_instance(self):
+        """
+        Take any predefined methods/handlers and insert them into Sanic JWT
+        """
+        # Initialize instance of the Authentication class
+        config = self.config
+        self.instance.auth = self.authentication_class(self.app, config=config)
+
+        if 'authenticate' not in self.kwargs:
+            raise exceptions.AuthenticateNotImplemented
+
+        for handler in handlers:
+            handler_name, _ = handler
+            if handler_name in self.kwargs:
+                method = self.kwargs.pop(handler_name)
+                setattr(self.instance.auth, handler_name, method)
+
+    def __load_configuration(self):
```

Comment:
I really like this new configuration method with its loading order. I'm just thinking what would be the usage of `**kwargs` in the real world, since it was not present in the previous versions? And oh, of course, this needs to be documented ...

Reply:
Oh forget about this question; I just realized that I'll have to use `**kwargs` in my application since some of them are loaded dynamically :grimacing:

Reply:
Also, for a simple implementation, kwargs may beore natural. Especially if you have multiple setups on blueprints. you could have all of them inherit from a single configuration class, but make any small tweaks in kwargs.

Reply:
Agreed.